platform/darwin/src/NSComparisonPredicate+MGLAdditions.mm

```diff
@@ -318,14 +318,32 @@ - (id)mgl_jsonExpressionObject {
         case NSNotEqualToPredicateOperatorType:
             op = @"!=";
             break;
+        case NSBetweenPredicateOperatorType: {
+            op = @"all";
+            NSArray *arguments = self.rightExpression.constantValue;
+            NSPredicate *leftHandPredicate = [NSComparisonPredicate predicateWithLeftExpression:arguments[0]
+                                                                                rightExpression:self.leftExpression
+                                                                                       modifier:NSAllPredicateModifier
+                                                                                           type:NSLessThanOrEqualToPredicateOperatorType
+                                                                                        options:0];
+            NSPredicate *rightHandPredicate = [NSComparisonPredicate predicateWithLeftExpression:self.leftExpression
+                                                                                 rightExpression:arguments[1]
+                                                                                        modifier:NSAllPredicateModifier
+                                                                                            type:NSLessThanOrEqualToPredicateOperatorType
+                                                                                         options:0];
+            return @[op, leftHandPredicate.mgl_jsonExpressionObject, rightHandPredicate.mgl_jsonExpressionObject];
+        }
+        case NSInPredicateOperatorType:
+            op = @"match";
+            break;
+        case NSContainsPredicateOperatorType:
+            op = @"has";
```

Comment:
A `CONTAINS` predicate should be translated into a `match` expression. For example, an NSExpression with the format string `{1, 2, 3, 4} CONTAINS a` should produce the following JSON object:

```json
[
  "match",
  ["get", "a"],
  [1, 2, 3, 4], true,
  false
]
```